src/components/catalog-parameters/catalog-parameters.controller.ts

```diff
@@ -110,9 +110,19 @@ export class CatalogParametersController implements angular.IController {
 
   private updateValueToHidden(value: any): any {
     if (_.isObject(value) || _.isArray(value)) {
-      return _.mapValues(value, (subValue: any) => {
-        return this.updateValueToHidden(subValue);
+      var returnValue: any = {};
+      _.each(_.keys(value), (key) => {
+        if (!_.find(this.ctrl.opaqueKeys, key)) {
```

Comment:
Doesn't this drop the opaque values since there's no `else`, or am I misunderstanding its purpose?